photutils/psf/tests/test_photometry.py

```diff
@@ -498,10 +499,68 @@ def test_aperture_radius_value_error():
                                     fitshape=7)
 
     intab = Table(data=[[1], [1]], names=['x_0', 'y_0'])
-    with pytest.raises(ValueError) as err:
+    with pytest.warns(AstropyUserWarning):
         basic_phot(image=image, init_guesses=intab)
 
-    assert 'aperture_radius is None' in str(err.value)
+
+def tophatfinder(image):
+    fluxes = np.unique(image[image > 0])
+    table = Table(names=['id', 'xcentroid', 'ycentroid', 'flux'],
+                  dtype=[int, float, float, float])
+    for n, f in enumerate(fluxes):
+        xs, ys = np.where(image == f)
+        x = np.mean(xs)
+        y = np.mean(ys)
+        table.add_row([int(n+1), x, y, f])
+
+    return table
+
+
+@pytest.mark.skipif('not HAS_SCIPY')
+def test_default_aperture_radius():
+    """
+    Test psf_photometry with non-Gaussian model, such that it raises a
+    warning about aperture_radius.
+    """
+
+    prf = np.zeros((7, 7), float)
+    prf[2:5, 2:5] = 1/9
+    prf = FittableImageModel(prf)
+
+    img = np.zeros((50, 50), float)
+    x0 = [20, 35, 38]
+    y0 = [5, 40, 20]
+    f0 = [100, 200, 50]
+    for x, y, f in zip(x0, y0, f0):
+        img[x+2:x+5, y+2:y+5] = f/9
+
+    basic_phot = BasicPSFPhotometry(group_maker=DAOGroup(2),
+                                    bkg_estimator=None, psf_model=prf,
+                                    fitshape=7)
+
+    intab = Table(data=[[19.6, 34.9, 37], [4.5, 40.1, 19.6]],
+                  names=['x_0', 'y_0'])
+    with pytest.warns(AstropyUserWarning):
```

Comment:
Here (and below), I think you should use `warns` with the `match` keyword (See https://docs.pytest.org/en/latest/warnings.html#warns) because you can never be sure that there isn't some other random astropyuserwarning you're accidentally catching if you don't check that the message is the right onw.

Reply:
Added appropriate and (hopefully) unique ``match`` keywords to the ``warns``, to avoid this potential confusion.